src/helpers/filterValidGroups.js

```diff
@@ -1,6 +1,6 @@
 const uniqBy = require('lodash/uniqBy')
 
 const filterValidGroups = groups => {
-  return uniqBy(groups, 'gid').filter(({ structure, gid }) => gid && structure)
+  return uniqBy(groups, 'uuid').filter(({ structure, gid }) => gid && structure)
```

Comment:
gid is the `displayed name` and `uuid` an uniq identifier?

Reply:
`gid` is not displayed no. It's an id used by toutatice, but it's not completely unique (they can be duplicated across structures). `uuid` is simething we construct in the connector and is really unique.